SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x-3.6.0/tsWxGTUI_Py3x-3.6.0/ManPages/tsManPagesLibGUI/tsWxAcceleratorEntry.man

Help on module tsWxAcceleratorEntry:

NAME
    tsWxAcceleratorEntry

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxAcceleratorEntry.py

DESCRIPTION
    tsWxAcceleratorEntry.py - Class to establish an object used by
    an application wishing to create an accelerator table
    (see wxAcceleratorTable).

CLASSES
    __builtin__.object
        AcceleratorEntry
    
    class AcceleratorEntry(__builtin__.object)
     |  A class used to define items in an wx.AcceleratorTable. wxPython
     |  programs can choose to use wx.AcceleratorEntry objects, but using
     |  a list of 3-tuple of integers (flags, keyCode, cmdID) usually
     |  works just as well. See __init__ for use of the tuple values.
     |  
     |  Methods defined here:
     |  
     |  FromString(self, str)
     |      Returns true if the given string correctly initialized this object.
     |  
     |  GetCommand(self)
     |      Get the AcceleratorEntry command ID.
     |  
     |  GetFlags(self)
     |      Get the AcceleratorEntry flags.
     |  
     |  GetKeyCode(self)
     |      Get the AcceleratorEntry keycode.
     |  
     |  IsOk(self)
     |  
     |  Set(self, flags, keyCode, cmdID)
     |      (Re)set the attributes of a wx.AcceleratorEntry.
     |  
     |  ToString(self)
     |      Returns a string representation for the this accelerator.
     |  
     |  __init__(self, flags=0, keyCode=0, cmdID=0)
     |      Construct a wx.AcceleratorEntry.
     |      
     |      flags - A bitmask of wx.ACCEL_ALT, wx.ACCEL_SHIFT, wx.ACCEL_CTRL,
     |      wx.ACCEL_CMD, or wx.ACCEL_NORMAL used to specify which modifier
     |      keys are held down. (type=int)
     |      
     |      keyCode - The keycode to be detected. (type=int)
     |      
     |      cmdID - The menu or control command ID to use for the accellerator
     |      event. (type=int)
     |  
     |  ----------------------------------------------------------------------
     |  Static methods defined here:
     |  
     |  Create(str)
     |      Create accelerator corresponding to the specified string, or None
     |      if it could not be parsed.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors defined here:
     |  
     |  Command
     |      Get the AcceleratorEntry command ID.
     |  
     |  Flags
     |      Get the AcceleratorEntry flags.
     |  
     |  KeyCode
     |      Get the AcceleratorEntry keycode.
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/0/2013'
    __header__ = '\n\ntsWxAcceleratorEntry, v1.0.1 (build 04/0/2013)... Ge...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxAcceleratorEntry, v1.0.1 (build 04/0/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxAcceleratorEntry'
    __version__ = '1.0.1'
    mainTitleVersionDate = 'tsWxAcceleratorEntry, v1.0.1 (build 04/0/2013)...

VERSION
    1.0.1

DATE
    04/0/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsDemoArchive/tsTutorialDeveloperSandbox2x/tsLibGUI/tsWxPkg/src/tsWxAcceleratorEntry.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  6:24:39 AM rsg>"
'''
tsWxAcceleratorEntry.py - Class to establish an object used by
an application wishing to create an accelerator table
(see wxAcceleratorTable).
'''
#################################################################
#
# File: tsWxAcceleratorEntry.py
#
# Purpose:
#
#    Class to establish an object used by an application wishing
#    to create an accelerator table (see wxAcceleratorTable).
#
# Usage (example):
#
#    # Import
#
#    from tsWxAcceleratorEntry import AcceleratorEntry
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    2012/07/29 rsg Corrected staticmethod Create to eliminate
#                   references to self.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxAcceleratorEntry'
__version__   = '1.0.1'
__date__      = '04/0/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

import tsWxGlobals as wx

#---------------------------------------------------------------------------

##DEBUG = True
DEBUG = wx.Debug_GUI_Launch | \
        wx.Debug_GUI_Progress | \
        wx.Debug_GUI_Termination | \
        wx.Debug_GUI_Exceptions

##VERBOSE = True
VERBOSE = wx.Debug_GUI_Configuration

#---------------------------------------------------------------------------

class AcceleratorEntry(object):
    '''
    A class used to define items in an wx.AcceleratorTable. wxPython
    programs can choose to use wx.AcceleratorEntry objects, but using
    a list of 3-tuple of integers (flags, keyCode, cmdID) usually
    works just as well. See __init__ for use of the tuple values.
    '''
    def __init__(self, flags=0, keyCode=0, cmdID=0):
        '''
        Construct a wx.AcceleratorEntry.

        flags - A bitmask of wx.ACCEL_ALT, wx.ACCEL_SHIFT, wx.ACCEL_CTRL,
        wx.ACCEL_CMD, or wx.ACCEL_NORMAL used to specify which modifier
        keys are held down. (type=int)

        keyCode - The keycode to be detected. (type=int)

        cmdID - The menu or control command ID to use for the accellerator
        event. (type=int)
        '''
        theClass = 'AcceleratorEntry'

        wx.RegisterFirstCallerClassName(self, theClass)

##        self.tsBeginClassRegistration(theClass, id)

        try:
            CmdID = int(cmdID)
            Flags = int(flags)
            KeyCode = int(keyCode)

            self.ts_AcceleratorEntry = (Flags, KeyCode, CmdID)
        except:
            self.ts_AcceleratorEntry = None

##        self.tsEndClassRegistration(theClass)

##    def __del__(self):
##        '''
##        '''
##        pass

    @staticmethod
    def Create(str):
        '''
        Create accelerator corresponding to the specified string, or None
        if it could not be parsed.
        '''
        try:

            (flags, keyCode, cmdID) = str.split('-')

            CommandID = int(cmdID)
            Flags = int(flags)
            KeyCode = int(keyCode)

            passed = True

        except:

            passed = False

        if passed:

            return (AcceleratorEntry(flags=Flags,
                                     keyCode=KeyCode,
                                     cmdID=CommandID))
        else:

            return (None)

    def FromString(self, str):
        '''
        Returns true if the given string correctly initialized this object.
        '''
        try:
            (flags, keyCode, cmdID) = str.split('-')

            CmdID = int(cmdID)
            Flags = int(flags)
            KeyCode = int(keyCode)

            self.ts_AcceleratorEntry = (Flags, KeyCode, CmdID)
            passed = True
        except:
            self.ts_AcceleratorEntry = None
            passed = False

        return (passed)

    def GetCommand(self):
        '''
        Get the AcceleratorEntry command ID.
        '''
        if self.ts_AcceleratorEntry is None:
            cmdID = None
        else:
            (flags, keyCode, cmdID) = self.ts_AcceleratorEntry

        return (cmdID)

    def GetFlags(self):
        '''
        Get the AcceleratorEntry flags.
        '''
        if self.ts_AcceleratorEntry is None:
            flags = None
        else:
            (flags, keyCode, cmdID) = self.ts_AcceleratorEntry

        return (flags)

    def GetKeyCode(self):
        '''
        Get the AcceleratorEntry keycode.
        '''
        if self.ts_AcceleratorEntry is None:
            keyCode = None
        else:
            (flags, keyCode, cmdID) = self.ts_AcceleratorEntry

        return (keyCode)

    def IsOk(self):
        '''
        '''
        if self.ts_AcceleratorEntry is None:
            return (False)
        else:
            return (True)

    def Set(self, flags, keyCode, cmdID):
        '''
        (Re)set the attributes of a wx.AcceleratorEntry.
        '''
        try:
            CmdID = int(cmdID)
            Flags = int(flags)
            KeyCode = int(keyCode)

            self.ts_AcceleratorEntry = (Flags, KeyCode, CmdID)
        except:
            self.ts_AcceleratorEntry = None

    def ToString(self):
        '''
        Returns a string representation for the this accelerator.
        '''
        textCmdID = str(self.GetCommand())
        textFlags = str(self.GetFlags())
        textKeyCode = str(self.GetKeyCode())
 
        text = '%s-%s-%s' % (textFlags, textKeyCode, textCmdID)
        return (text)

    Command = property(GetCommand)
    Flags = property(GetFlags)
    KeyCode = property(GetKeyCode)

#---------------------------------------------------------------------------
 
if __name__ == '__main__':

    print(__header__)
